File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s)!=len(t):
            return False
        hashmap_s=defaultdict(int)
        hashmap_t=defaultdict(int)
        for i in range(len(s)):
            hashmap_s[s[i]]+=1
            hashmap_t[t[i]]+=1
        return hashmap_s==hashmap_t

File: 242-valid-anagram/valid-anagram.java
class Solution {

    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        HashMap<Character, Integer> hashmapS = new HashMap<Character, Integer>();
        HashMap<Character, Integer> hashmapT = new HashMap<Character, Integer>();
        System.out.println(hashmapS);
        System.out.println(hashmapT);
        for (int i = 0; i < s.length(); i++) {
            hashmapS.put(s.charAt(i), hashmapS.getOrDefault(s.charAt(i), 0) + 1);
            hashmapT.put(t.charAt(i), hashmapT.getOrDefault(t.charAt(i), 0) + 1);
        }
        System.out.println(hashmapS);
        System.out.println(hashmapT);
        return hashmapS.equals(hashmapT);
    }
}
